stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -242,6 +242,27 @@ public class ChatClient internal constructor(
         return createInitListenerCall { initListener -> setUser(user, tokenProvider, initListener) }
     }
 
+    /**
+     * Initializes [ChatClient] for a specific user and a given [userToken].
+     * Caution: This method doesn't establish connection to the web socket, you should use [connectUser] instead.
+     *
+     * This method initializes [ChatClient] to allow the use of Stream REST API client.
+     * Moreover, it warms up the connection, and sets up notifications.
+     *
+     * @param user the user to set
+     * @param userToken the user token
+     */
+    public fun setUserWithoutConnecting(user: User, userToken: String) {
```

Comment:
> setUserWithoutConnecting

We should talk to other teams and see what they call methods like this, or if they have them available at all. We just had to do a bunch of renaming around `setUser`, would be good to be synchronized on new APIs around here.

Reply:
Right, I'll start a cross-team discussion about this.

Reply:
Since we don't have consistent APIs at the moment (especially compared to iOS) and other platforms don't need this feature. My suggestion is to add the function as it is and refactor it later along with iOS. I believe having it as a simple separate function would be easy to refactor in the future.

Reply:
Can't we hide this method from our public API?
At the end it is not a method that our user will use, right?

Reply:
But we need an access to this method in the `offline` module.

Reply:
Sounds like `@InternalStreamChatApi`

Reply:
Added!